Tests/NIOPosixTests/BootstrapTest.swift

```diff
@@ -680,6 +680,30 @@ class BootstrapTest: XCTestCase {
                                 .wait())
         }
     }
+
+    // There was a bug where file handle ownership was not released when creating pipe channels failed.
+    func testReleaseFileHandleOnOwningFailure() {
+        struct NIOPipeBootstrapHooksChannelFail: NIOPipeBootstrapHooks {
+            func makePipeChannel(eventLoop: NIOPosix.SelectableEventLoop, inputPipe: NIOCore.NIOFileHandle?, outputPipe: NIOCore.NIOFileHandle?) throws -> NIOPosix.PipeChannel {
+                throw IOError(errnoCode: EBADF, reason: "testing")
+            }
+        }
+
+        let sock = socket(AF_LOCAL, SOCK_STREAM, 0)
+        defer {
+            close(sock)
+        }
```

Comment:
I don't think we're using this socket anywhere.

Reply:
Used on line 704 - `let channelFuture = bootstrap?.takingOwnershipOfDescriptor(inputOutput: **sock**)`